service/collector/subnet.go

```diff
@@ -256,8 +298,25 @@ func (e *Subnet) getSubnetInfoFromAPI(ctx context.Context, awsClients clientaws.
 		if subnet.Tags[key.TagInstallation] != e.installationName {
 			continue
 		}
+		subnet.AvailableIPPercentage, err = getAvailableIPPercentage(subnet.Tags["CidrBlock"], subnet.AvailableIPs)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
 		subnets = append(subnets, subnet)
 	}
 	res.Subnets = subnets
 	return &res, nil
 }
+
+func getAvailableIPPercentage(cidr string, availableIps int64) (float64, error) {
+	parts := strings.Split(cidr, "/")
+	if len(parts) != 2 {
+		return 0, microerror.Maskf(parsingFailedError, "Can not parse CIDR.")
+	}
+	if n, err := strconv.Atoi(parts[1]); err == nil {
+		totalIPs := math.Pow(2, float64(32-n)) - 2
```

Comment:
what is the magical number 2 here?

Reply:
As far as I remember, 2 IPs in the network are always reserved for the host network and broadcast network. And you are right, I should add the comment.

Reply:
That is true, but on AWS there are more reserved IP, the  first usable IP is used for VPC router as the default route , the second usable IP is for the DNS server and the third usable IP is reserved by AWS for future use, so basically 5 ips are not usable from network on AWS

its somewhere here https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Subnets.html

Reply:
Thank you, that is critical information. :)